pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go

```diff
@@ -113,3 +125,99 @@ func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
 		dswp.desiredStateOfWorld.DeletePod(dswPodUID, dswPodToAdd.VolumeName, dswPodToAdd.NodeName)
 	}
 }
+
+func (dswp *desiredStateOfWorldPopulator) isPodTerminated(pod *v1.Pod) bool {
```

Comment:
This will sound nit picky but since the line `return pod.Status.Phase == v1.PodFailed || pod.Status.Phase == v1.PodSucceeded || (pod.DeletionTimestamp != nil && volumehelper.NotRunning(pod.Status.ContainerStatuses))` is common and is non-trivial (meaning there could be a bug in it), I would prefer to have this moved to a common package. That way if there is a bug in this statement, it will be fixed in one place.